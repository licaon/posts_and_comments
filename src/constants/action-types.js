export const INIT = 'INIT';

export const START_LOADING_DATA = 'START_LOADING_DATA';
export const END_LOADING_DATA = 'END_LOADING_DATA';

export const GET_POSTS = 'GET_POSTS';
export const ADD_POST = 'ADD_POST';
export const DELETE_POST = 'DELETE_POST';

export const ADD_SEARCH_FILTER = 'ADD_SEARCH_FILTER';
export const ADD_SELECT_FILTER = 'ADD_SELECT_FILTER';

export const GET_USERS = 'GET_USERS';
export const ADD_USER = 'ADD_USER';
