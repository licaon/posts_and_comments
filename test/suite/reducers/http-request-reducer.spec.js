import { START_LOADING_DATA, END_LOADING_DATA } from 'src/constants/action-types';
import HttpRequestReducer from 'src/reducers/http-request-reducer';

describe('HttpRequestReducer', () => {
  it(`should update reducer correctly on ${START_LOADING_DATA}`, (done) => {
    expect(HttpRequestReducer(
      undefined,
      {
        type: START_LOADING_DATA,
      },
    )).toMatchSnapshot();
    done();
  });
  it(`should update reducer correctly on ${END_LOADING_DATA}`, (done) => {
    expect(HttpRequestReducer(
      {
        loading: true,
      },
      {
        type: END_LOADING_DATA,
      },
    )).toMatchSnapshot();
    done();
  });
  it(`should return reducer correctly on unknown action`, (done) => {
    expect(HttpRequestReducer(
      undefined,
      { type: 'unknown' },
    )).toMatchSnapshot();
    done();
  });
});
