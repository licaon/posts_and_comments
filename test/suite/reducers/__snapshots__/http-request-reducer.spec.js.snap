// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HttpRequestReducer should return reducer correctly on unknown action 1`] = `
Object {
  "loading": false,
}
`;

exports[`HttpRequestReducer should update reducer correctly on END_LOADING_DATA 1`] = `
Object {
  "loading": false,
}
`;

exports[`HttpRequestReducer should update reducer correctly on START_LOADING_DATA 1`] = `
Object {
  "loading": true,
}
`;
